src/language.rs

```diff
@@ -24,6 +24,14 @@ impl WordList {
     pub fn get_word(&self, bits: Bits11) -> &'static str {
         self.inner[bits.bits() as usize]
     }
+
+    pub fn get_words_by_prefix(&self, prefix: &str) -> Vec<&'static str> {
+        self.inner
+            .iter()
+            .filter(|word| word.starts_with(prefix))
```

Comment:
Since the dictionary is sorted alphabetically, you can turn worst case O(n) into for the first element O(log n), and stop on first miss. You also can just borrow a slice from the list instead of allocating a new Vec:

```rust
pub fn get_words_by_prefix(&self, prefix: &str) -> &[&'static str] {
    let start = match self.inner.binary_search(prefix) {
        Ok(idx) => idx,
        Err(idx) => idx,
    };
    let end = start + self.inner[start..].iter().take_while(|word| word.starts_with(prefix)).count();

    &self.inner[start..end]
}
```

Reply:
Thanks, I forced pushed this version into the PR branch.